native/core/src/execution/utils.rs

```diff
@@ -96,6 +99,22 @@ impl SparkArrowConvert for ArrayData {
 
         Ok((array as i64, schema as i64))
     }
+
+    /// Move this ArrowData to pointers of Arrow C data interface.
+    fn move_to_spark(&self, array: i64, schema: i64) -> Result<(), ExecutionError> {
+        let jvm_array = unsafe { std::ptr::replace(array as *mut FFI_ArrowArray, FFI_ArrowArray::new(self)) };
+        let jvm_schema = unsafe {
+            std::ptr::replace(
+                schema as *mut FFI_ArrowSchema,
+                FFI_ArrowSchema::try_from(self.data_type())?,
+            )
+        };
+
+        std::mem::forget(jvm_array);
+        std::mem::forget(jvm_schema);
```

Comment:
Saw SIGSEGV in CI on ubuntu platform only (I cannot reproduce locally on Mac). Trying this to see if it is related.

Reply:
Hmm, seems the SIGSEGV failures was caused by dropping the array/schema structures returned by `replace`. 🤔

Reply:
Trying a fix.

Reply:
If the base structures allocated by JVM cannot be safely dropped, we may use `std::ptr::write(array as *mut FFI_ArrowArray, FFI_ArrowArray::new(self))` to simply overwrite the base structure, without moving the original structure out.

Reply:
I fixed this. Please see the latest commit.